src/_tkagg.cpp

```diff
@@ -324,10 +324,12 @@ void load_tkinter_funcs(void)
 
 exit:
     // We don't need to keep a reference open as the main program & tkinter
-    // have been imported.  Use a non-short-circuiting "or" to try closing both
-    // handles before handling errors.
-    if ((main_program && dlclose(main_program))
-        | (tkinter_lib && dlclose(tkinter_lib))) {
+    // have been imported.  Try to close each library separately (otherwise the
+    // second dlclose could clear a dlerror from the first dlclose).
+    if (main_program && dlclose(main_program)) {
+        PyErr_SetString(PyExc_RuntimeError, dlerror());
+    }
+    if (tkinter_lib && dlclose(tkinter_lib)) {
         PyErr_SetString(PyExc_RuntimeError, dlerror());
```

Comment:
I couldn't find an explicit note about it, but do you know if Python is fine with calling this a second time in a row (without clearing the error) if both calls fail?

Reply:
Good question; added a check for that.